ape_etherscan/explorer.py

```diff
@@ -34,16 +35,39 @@ def get_contract_type(self, address: AddressType) -> Optional[ContractType]:
             # Handle non-checksummed addresses
             address = self.conversion_manager.convert(str(address), AddressType)
 
-        client = self._client_factory.get_contract_client(address)
-        source_code = client.get_source_code()
+        contract_client = self._client_factory.get_contract_client(address)
+        source_code = contract_client.get_source_code()
         if not (abi_string := source_code.abi):
             return None
 
         try:
             abi = json.loads(abi_string)
         except JSONDecodeError as err:
-            logger.error(f"Error with contract ABI: {err}")
-            return None
+            contract_creation = contract_client.get_creation_data()
+            if not contract_creation:
+                return None
+
+            if not hasattr(contract_creation[0], "txHash"):
+                return None
+
+            tx_hash = contract_creation[0].txHash
+            if not tx_hash:
+                return None
+
+            proxy_client = self._client_factory.get_proxy_client(tx_hash)
+            transaction_by_hash = proxy_client.get_transaction_by_hash()
```

Comment:
Instead of making a new client, you could do `transaction_by_hash = self.provider.ge_receipt(tx_hash)`